py/genv/runners/runner.py

```diff
@@ -24,6 +24,17 @@ async def run(self, *args: str, stdin: Optional[str] = None, sudo: bool = False)
 
         return process, stdout, stderr
 
+    async def run_to_success(self, *args: str, stdin: Optional[str] = None, sudo: bool = False) -> str:
```

Comment:
I think it would be better to keep it as a seperate function because of the of the return arguments - run returns process, stdout, stdin. run_to_success returns only stdout. 
I think this is better for the sake of simplicity.